src/verify_multiple_aggregate_signatures.cc

```diff
@@ -0,0 +1,207 @@
+                return scope.Escape(env.Undefined());
+            case blst_ts::BLST_TS_ERROR::HAS_NATIVE_ERROR:
+                Napi::Error::New(env, error_msg).ThrowAsJavaScriptException();
+                return scope.Escape(env.Undefined());
+            default:
+            case blst_ts::BLST_TS_ERROR::INVALID:
+                return scope.Escape(Napi::Boolean::New(env, false));
+        }
+    } catch (...) {
+        return Napi::Boolean::New(env, false);
+    }
+}
+
+class VerifyMultipleAggregateSignaturesWorker : public Napi::AsyncWorker {
+   public:
+    VerifyMultipleAggregateSignaturesWorker(const Napi::CallbackInfo &info)
+        : Napi::
+              AsyncWorker{info.Env(), "VerifyMultipleAggregateSignaturesWorker"},
+          deferred{Env()},
+          has_error{false},
+          _module{Env().GetInstanceData<BlstTsAddon>()},
+          _ctx{new blst::Pairing(true, _module->dst)},
+          _sets{},
+          _sets_ref{Napi::Persistent(info[0])},
+          _is_invalid{false},
+          _result{false} {
+        try {
+            blst_ts::BLST_TS_ERROR error =
+                prepare_verify_multiple_aggregate_signatures(_sets, info);
+            switch (error) {
+                case blst_ts::BLST_TS_ERROR::SUCCESS:
+                    return;
+                case blst_ts::BLST_TS_ERROR::JS_ERROR_THROWN:
+                    has_error = true;
+                    return;
+                default:
+                case blst_ts::BLST_TS_ERROR::INVALID:
+                    _is_invalid = true;
+            }
+        } catch (...) {
+            _is_invalid = true;
+        }
+    }
+
+    /**
+     * GetPromise associated with deferred for return to JS
+     */
+    Napi::Promise GetPromise() { return deferred.Promise(); }
+
+   protected:
+    void Execute() {
+        if (_is_invalid) {
+            return;
+        }
+        std::string error_msg{};
+        blst_ts::BLST_TS_ERROR error = verify_multiple_aggregate_signatures(
+            _result, error_msg, _module, _ctx, _sets);
+        if (error == blst_ts::BLST_TS_ERROR::HAS_NATIVE_ERROR) {
+            SetError(error_msg);
+        }
+    }
+    void OnOK() { deferred.Resolve(Napi::Boolean::New(Env(), _result)); }
+    void OnError(const Napi::Error &err) { deferred.Reject(err.Value()); }
+
+   public:
+    Napi::Promise::Deferred deferred;
+    bool has_error;
+
+   private:
+    BlstTsAddon *_module;
+    std::unique_ptr<blst::Pairing> _ctx;
+    std::vector<SignatureSet> _sets;
+    Napi::Reference<Napi::Value> _sets_ref;
+    bool _is_invalid;
+    bool _result;
+};
+
+Napi::Value AsyncVerifyMultipleAggregateSignatures(
+    const Napi::CallbackInfo &info) {
+    VerifyMultipleAggregateSignaturesWorker *worker =
+        new VerifyMultipleAggregateSignaturesWorker(info);
+    if (worker->has_error) {
+        delete worker;
+        return info.Env().Undefined();
+    }
+    worker->Queue();
+    return worker->GetPromise();
```

Comment:
Who will delete `worker`?

Reply:
The node runtime does that after the async work calls back into JS (for both valid and invalid cases)

Reply:
It would make sense to add a comment there, then, for future readers.